src/main/java/Duke.java

```diff
@@ -1,10 +1,272 @@
+            try{
+                if (command.equalsIgnoreCase("bye")) {
+                    System.out.println("Bye. Hope to see you again soon!");
+                    break;
+                } else if (command.equalsIgnoreCase("list")) {
+                    System.out.println("Here are the tasks in your list:");
+                    for (int i = 0; i < tasks.size(); i++) {
+                        System.out.println((i + 1) + ". " + tasks.get(i));
+                    }
+                } else if (command.startsWith("todo")) {
+                    String description = command.substring(5).trim();
+                    if (description.isEmpty()) {
+                        throw new DukeException("The description of a todo cannot be empty.");
+                    }
+                    tasks.add(new Todo(description));
+                    System.out.println("Got it. I've added this task:\n  " + tasks.get(tasks.size() - 1));
+                    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                } else if (command.startsWith("deadline")) {
+                    // Parse the date and time in the format d/M/yyyy HHmm
+                    String[] parts = command.split(" /by ");
+                    if (parts.length < 2) {
+                        throw new DukeException("Deadline command must include a date.");
+                    }
+                    String description = parts[0].substring(9).trim();
+                    LocalDateTime dateTime = LocalDateTime.parse(parts[1], DateTimeFormatter.ofPattern("d/M/yyyy HHmm"));
+
+                    tasks.add(new Deadline(description, dateTime));
+                    System.out.println("Got it. I've added this task:\n  " + tasks.get(tasks.size() - 1));
+                    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                } else if (command.startsWith("event")) {
+                    String description = command.substring(6, command.indexOf("/from")).trim();
+                    String from = command.substring(command.indexOf("/from") + 6, command.indexOf("/to")).trim();
+                    String to = command.substring(command.indexOf("/to") + 4).trim();
+                    tasks.add(new Event(description, from, to));
+                    System.out.println("Got it. I've added this task:\n  " + tasks.get(tasks.size() - 1));
+                    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                } else if (command.startsWith("mark")) {
+                    int index = Integer.parseInt(command.split(" ")[1]) - 1;
+                    if (index >= 0 && index < tasks.size()) {
+                        tasks.get(index).markDone();
+                        System.out.println("Nice! I've marked this task as done:\n  " + tasks.get(index));
+                    } else {
+                        System.out.println("Invalid task index.");
```

Comment:
would it be useful if instead of printing invalid task index, the size of the tasklist is also returned in this else case to let user know that for example there is only 3 tasks in the tasklist thats why their input 4 is invalid. Something like "Invalid task index, the task list has " + tasklist.size() + "tasks. "